pkg/services/accesscontrol/evaluator.go

```diff
@@ -0,0 +1,178 @@
+package accesscontrol
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"strings"
+
+	"github.com/gobwas/glob"
+)
+
+type Evaluator interface {
+	// Evaluate permissions that are grouped by action
+	Evaluate(permissions map[string]map[string]struct{}) (bool, error)
+	// Inject params into the evaluator's templated scopes. e.g. "settings:" + eval.Parameters(":id") and returns a new Evaluator
+	Inject(params map[string]string) (Evaluator, error)
+	// String returns a string representation of permission required by the evaluator
+	String() string
+}
```

Comment:
Let's have a separate issue to discuss the terminology and API used here, but since that's just an internal refactoring, we don't have to do that in this PR.